packages/frontend/src/components/common/ShareSpaceModal/ShareSpaceUserList.tsx

```diff
@@ -529,7 +529,19 @@ export const ShareSpaceUserList: FC<ShareSpaceUserListProps> = ({
     );
 
     const accessByType = useMemo<SpaceAccessByType>(() => {
-        return space.access.reduce<SpaceAccessByType>(
+        const seenUsers = new Set<string>();
+
+        // don't show duplicate users - this can happen if a user belongs to multiple groups
+        const deduplicateUsers = (spaceShares: SpaceShare[]) =>
```

Comment:
I believe this will cause issues if they have different roles in each group. Also, should the `seenUsers` be inside the function or is intentional that is filtering across project/org/direct?

Reply:
The org and project collapse section should list all the users that inherit access from them.
The users list at the bottom should list all users with access to the space. That includes all users with direct access and inherited access from org/project/group. 
At the moment, we dedupe across all the sections, causing users to show in only 1 of those places.

I don't understand why users with inherited access are coming up with `hasDirectAccess = true`. Something is wrong up the chain :/

Reply:
Updated it to only dedup users + show their highest role 🤙